packages/tool-cache/src/tool-cache.ts

```diff
@@ -83,23 +83,21 @@ export async function downloadTool(
 
       const file: NodeJS.WritableStream = fs.createWriteStream(dest)
       file.on('open', async () => {
-        try {
-          const stream = response.message.pipe(file)
-          stream.on('close', () => {
-            core.debug('download complete')
-            resolve(dest)
-          })
-        } catch (err) {
-          core.debug(
-            `Failed to download from "${url}". Code(${response.message.statusCode}) Message(${response.message.statusMessage})`
-          )
+        const readStream = response.message
+        readStream.pipe(file)
+        readStream.on('error', (err: Error) => {
```

Comment:
This error handler for the response body stream is new.

I was able to confirm it was missing by forcing an error: i hooked up to the on data, and then called response.message.emit('error', ...). And an unhandled error bubbled.